service/db.go

```diff
@@ -46,3 +54,62 @@ func (db *LevelDB) Delete(key []byte) error {
 func (db *LevelDB) Iterator() iterator.Iterator {
 	return db.DB.NewIterator(nil, db.ro)
 }
+
+type ProofsDatabase struct {
+	db     *leveldb.DB
+	proofs <-chan shared.ProofMessage
+}
+
+func (db *ProofsDatabase) Get(ctx context.Context, roundID string) (*shared.ProofMessage, error) {
+	data, err := db.db.Get([]byte(roundID), nil)
+	if err != nil {
+		return nil, fmt.Errorf("get proof for %s from DB: %w", roundID, err)
+	}
+
+	proof := &shared.ProofMessage{}
+	if _, err := proof.DecodeScale(scale.NewDecoder(bytes.NewReader(data))); err != nil {
+		return nil, fmt.Errorf("failed to get deserialize proof: %w", err)
+	}
+	return proof, nil
+}
+
+func NewProofsDatabase(dbPath string, proofs <-chan shared.ProofMessage) (*ProofsDatabase, error) {
+	db, err := leveldb.OpenFile(dbPath, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open database @ %s: %w", dbPath, err)
+	}
+
+	return &ProofsDatabase{db, proofs}, nil
+}
+
+func (db *ProofsDatabase) Run(ctx context.Context) error {
+	logger := log.AppLog.WithName("proofs-db")
+	for {
+		select {
+		case proof := <-db.proofs:
+			serialized, err := serializeProofMsg(proof)
+			if err != nil {
+				logger.With().Error("failed serializing proof", log.Err(err))
```

Comment:
stop work here, or log fatal, i prefer the latter as this should not happen.

Reply:
I will return an error. The owner of DB will have a choice of what to do next.